src/Plugin/Field/FieldWidget/OgComplex.php

```diff
@@ -26,13 +30,183 @@
+
+    // Adding the other groups widget.
+    if ($this->isGroupAdmin()) {
+      $parent_form['other_groups'] = $this->otherGroupsWidget($items, $form_state);
+    }
+
+    return $parent_form;
+  }
+
+  /**
+   * Adding the other groups widget to the form.
+   *
+   * @param $elements
+   *   The widget array.
+   */
+  protected function otherGroupsWidget(FieldItemListInterface $items, FormStateInterface $form_state) {
+    if ($this->fieldDefinition->getTargetEntityTypeId() == 'user') {
+      $description = $this->t('As groups administrator, associate this user with groups you do <em>not</em> belong to.');
+    }
+    else {
+      $description = $this->t('As groups administrator, associate this content with groups you do <em>not</em> belong to.');
+    }
+
+    $field_wrapper = Html::getClass($this->fieldDefinition->getName()) . '-add-another-group';
+
+    $elements = [
+      '#type' => 'container',
+      '#tree' => TRUE,
+      '#title' => $this->t('Other groups'),
+      '#description' => $description,
+      '#prefix' => '<div id="' . $field_wrapper . '">',
+      '#suffix' => '</div>',
+      '#cardinality' => FieldStorageDefinitionInterface::CARDINALITY_UNLIMITED,
+      '#cardinality_multiple' => TRUE,
+      '#theme' => 'field_multiple_value_form',
+      '#field_name' => $this->fieldDefinition->getName(),
+      '#max_delta' => 1,
+    ];
+
+    $elements['add_more'] = [
+      '#type' => 'button',
+      '#value' => $this->t('Add another item'),
+      '#name' => 'add_another_group',
+      '#ajax' => [
+        'callback' => [$this, 'addMoreAjax'],
+        'wrapper' => $field_wrapper,
+        'effect' => 'fade',
+      ],
+    ];
+
+    $start_key = 0;
+
+    $target_type = $this->fieldDefinition->getTargetEntityTypeId();
+    $entity_group_ids = array_map(function($entity) {
+      return $entity->id();
+    }, $items->referencedEntities());
+
+    $other_groups = Og::getEntityGroups('user', \Drupal::currentUser()->id());
+    $other_groups_target_type = isset($other_groups[$target_type]) ? $other_groups[$target_type] : [];
+    $other_groups_weight_delta = round(count($other_groups) / 2);
+
+    foreach ($other_groups_target_type as $other_group) {
+      if (!in_array($other_group->id(), $entity_group_ids)) {
+        continue;
+      }
+      $elements[$start_key] = $this->otherGroupsSingle($start_key, $other_group, $other_groups_weight_delta);
+      $start_key++;
+    }
+
+    if (!$form_state->get('other_group_delta')) {
+      $form_state->set('other_group_delta', $start_key);
+    }
+
+    // Get the trigger element and check if this the add another item button.
+    $trigger_element = $form_state->getTriggeringElement();
+
+    if ($trigger_element['#name'] == 'add_another_group') {
+      // Increase the number of other groups.
+      $delta = $form_state->get('other_group_delta') + 1;
+      $form_state->set('other_group_delta', $delta);
+    }
+
+    // Add another auto complete field.
```

Comment:
we are hardcoing autocomplete here, which we shouldn't.

In D7 we had these options in the field settings.

![groups_audience___site-install-2](https://cloud.githubusercontent.com/assets/125707/11117977/071da510-8947-11e5-832c-731439dc721b.jpg)

(@damiankloip if you prefer we can do it in a follow up PR)